lib/io.js

```diff
@@ -650,6 +650,9 @@ DatumWriter.prototype = {
                     ) {
                 schemaIndex = i;
                 break;
+            } else if (writersSchema.schemas[i].type === 'map' && writersSchema.schemas[i].validate(writersSchema.schemas[i].type, datum)) {
+                schemaIndex = i;
+                break;
```

Comment:
I'm not sure. This and most of the other changes were all in https://github.com/thecsyco/node-avro-io/pull/1, so I kept them. You'd probably be more qualified at this point to edit them down if you think it necessary.